omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,301 @@
+			}
+			if (!StringUtils.isBlank(minLength)) {
+				if(StringUtils.isNumeric(minLength)) {
+					identifierSource.setMinLength(Integer.parseInt(minLength));
+				}else {
+		        		exception.add("min length");
+		        }
+	        }
+			if (!StringUtils.isBlank(maxLength)) {
+				if(StringUtils.isNumeric(maxLength)) {
+					identifierSource.setMaxLength(Integer.parseInt(maxLength));
+				}else {
+		        		exception.add("max length");
+		        }
+	        }
+			if(exception.toString().length() > 2) {
+				throw new ConversionException("The values of the following inputs are missing or invalid: " + exception.toString());
+			}
+			
+			identifierSource.setIdentifierType(patientIdentifierType);
+			identifierSource.setName(name);
+			identifierSource.setDescription(description);
+			identifierSource.setBaseCharacterSet(baseCharacterSet);
+			identifierSource.setFirstIdentifierBase(firstIdentifierBase);
+			
+			savedIdentifierSource =  save(identifierSource);	
+		}
+		else if(identifierSourceType.equals("RemoteIdentifierSource")) {
+			
+			RemoteIdentifierSource identifierSource = new RemoteIdentifierSource();
+			String username = validateInput(postBody.get("username"));
+			String password = validateInput(postBody.get("password"));
+			String url = validateInput(postBody.get("url"));
+
+			if (StringUtils.isBlank(url)) { 
+				exception.add("url"); 
+			}
+			if (!StringUtils.isBlank(username)) { 
+				identifierSource.setUser(username); 
+			}
+			if (!StringUtils.isBlank(password)) { 
+				if(StringUtils.isBlank(username)) { 
+					exception.add("username");
+					}
+				else { 
+					identifierSource.setPassword(password); 
+					}
+				}
+			if(exception.toString().length() > 2) {
+				throw new ConversionException("The values of the following inputs are missing or invalid: " + exception.toString());
+			}
+			
+			identifierSource.setIdentifierType(patientIdentifierType);
+			identifierSource.setName(name);
+			identifierSource.setUrl(url);
+			identifierSource.setDescription(description);
+	
+			savedIdentifierSource =  save(identifierSource);
+			
+		}
+		else if(identifierSourceType.equals("IdentifierPool")){
+			
+			IdentifierPool identifierSource = new IdentifierPool();
+			String batchSize = validateInput(postBody.get("batchSize"));
+			String minPoolSize = validateInput(postBody.get("minPoolSize"));
+			String sourceUuid = validateInput(postBody.get("sourceUuid"));
+			String sequential = validateInput(postBody.get("sequential"));
+			String refillWithScheduledTask = validateInput(postBody.get("refillWithScheduledTask"));
+			IdentifierSource poolSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid);
+			
+			if (!StringUtils.isBlank(batchSize)) {
+				if(StringUtils.isNumeric(batchSize)) {
+					identifierSource.setBatchSize(Integer.parseInt(batchSize));
+				}else {
+		        		exception.add("batch size");
+		        }
+	        }
+			if (!StringUtils.isBlank(minPoolSize)) {
```

Comment:
Is the above line having the correct indention?

Reply:
My IDE (Eclipse) is showing that everything is ok. Let me try a linter.

Reply:
Switched to Netbeans.